File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
import java.util.*;

class Solution {
    public long solution(int r1, int r2) {
        
        long answer = 0;
        
        for(int i=1; i<r2; i++){
            // x좌표기준 반지름이 s2가 되는 점의 좌표개수
            int max = (int)Math.sqrt(Math.pow(r2,2)-Math.pow(i,2));
            int min = (int)Math.ceil(Math.sqrt(Math.pow(r1,2)-Math.pow(i,2)));
            answer += max-min+1;
        }
        
        answer *= 4;
        answer += 4;
        
        return answer;
    }
}

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.js
function solution(r1, r2) {
    
    var answer = 0;
    
    for(let i=1; i<=r2; i++){
        var max = Math.floor(Math.sqrt(Math.pow(r2,2) - Math.pow(i,2)))
        var min = Math.ceil(Math.sqrt(Math.pow(r1,2) - Math.pow(i, 2))) || 0
        var cnt = max - min + 1;
        answer += cnt*4;
    }
    
    //answer += 4;
   
    return answer;
}